src/v/raft/group_configuration.cc

```diff
@@ -494,8 +537,16 @@ adl<raft::group_configuration>::from(iobuf_parser& p) {
     if (version > 0) {
         revision = adl<model::revision_id>{}.from(p);
     }
-    return raft::group_configuration(
+    auto cfg = raft::group_configuration(
       std::move(brokers), std::move(current), revision, std::move(old));
+    if (version > 2) {
```

Comment:
this is a merge skew, decommissioning requires nodes that all replicas from nodes that are decommissioned are moved to another node. If this is not true the decommissioning process would never end. Configuration is correctly replicated so the node should be able to read it after it will be updated.

Reply:
what i'm saying is that if node decommissioning is in progress, and then controller leadership transfers to a node running a version of the software that doesn't contain logic for node decommissioning, then what happens? more generally, how do we know there are no problematic scenarios, particularly in the context of a mixed version, dynamic cluster?